tests/rag_answers/custom_deepeval/metrics/test_factual_correctness.py

```diff
@@ -0,0 +1,415 @@
+                "mocked_prompt", schema=FactClassificationResult
+            )
+
+        @pytest.mark.asyncio
+        async def test_non_native_model_returns_recoverable_json(
+            self, mock_non_native_model: Mock, test_case: LLMTestCase
+        ):
+            # fallback case: schema call fails, raw JSON string returned
+            fixable_json = (
+                '{"classified_facts": {"TP": ["fact1"], "FP": ["fact2"], "FN": []}}'
+            )
+
+            # first call raises TypeError, second returns fixable string
+            mock_non_native_model.a_generate = AsyncMock(
+                side_effect=[
+                    TypeError("schema parse failed"),
+                    fixable_json,
+                ]
+            )
+
+            metric = FactualCorrectnessMetric(model=mock_non_native_model)  # type: ignore
+            _ = await metric.a_measure(test_case)
+
+            # verify a_generate was called twice:
+            # 1. with schema
+            # 2. without schema (fallback)
+            assert mock_non_native_model.a_generate.call_count == 2
+            first_call_args = mock_non_native_model.a_generate.call_args_list[
+                0
+            ]  # should include 'schema'
+            second_call_args = mock_non_native_model.a_generate.call_args_list[
+                1
+            ]  # should not include 'schema'
+
+            assert "schema" in first_call_args.kwargs
+            assert "schema" not in second_call_args.kwargs
+
+        @pytest.mark.asyncio
+        async def test_non_native_model_returns_unparsable_json(
+            self, mock_non_native_model: Mock, test_case: LLMTestCase
+        ):
+            mock_non_native_model.a_generate = AsyncMock(
+                side_effect=Exception("unexpected")
+            )
+            metric = FactualCorrectnessMetric(model=mock_non_native_model)  # type: ignore
+            _ = await metric.a_measure(test_case=test_case)
+
+            assert isinstance(metric.confusion_matrix, ClassifiedFacts)
+
+            assert metric.confusion_matrix.TP == []
+            assert metric.confusion_matrix.FP == []
+
+    def test_measure_raises_not_implemented(
+        self, mock_native_model: Mock, test_case: LLMTestCase
+    ):
+        metric = FactualCorrectnessMetric(model=mock_native_model)
+
+        with pytest.raises(
+            NotImplementedError, match="Synchronous evaluation is not supported"
+        ):
+            metric.measure(test_case)
+
+    def test_is_successful(self, mock_native_model: Mock):
+        metric = FactualCorrectnessMetric(
+            model=mock_native_model, threshold=0.7, include_reason=True
+        )
+
+        metric.success = True
+        assert metric.is_successful() is True
+
+        metric.success = False
+        assert metric.is_successful() is False
+
+        metric.error = "Some error"
+        assert metric.is_successful() is False
```

Comment:
It'd be better practice to test these based on the input into metric rather than changing the class state directly. 

I'd suggest you could use threshold as the means to check basic True and False. For the error state I'm a bit confused how we can fall into that so will leave a comment there.